enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -4124,6 +4124,9 @@ Constant *GradientUtils::GetOrCreateShadowFunction(
     return ConstantExpr::getPointerCast(GV, fn->getType());
   }
   }
+  llvm::errs() << "unhandled mode\n";
```

Comment:
why is this necessary? We should already handle all the cases.

Reply:
@ZuseZ4 remove the llvm errs warning and change the llvm_unreachable text to "Illegal state: unknown mode for GetOrCreateShadowFunction

@tgymnich my guess is that some compilers cannot prove that we handle all cases of the enum, and this is seen as a path without a return/unreachable.

Reply:
Indeed, for clang 14.0.6 it's not enough to handle all valid enum states in the places I updated.
At least this fixed the warning for it.

I'll adjust the message and correct tablegen tmrw.